conans/client/graph/compatibility.py

```diff
@@ -120,4 +121,7 @@ def _compatible_infos(conanfile, compatibles):
                 if options:
                     compat_info.options.update(options_values=OrderedDict(options))
                 result.append(compat_info)
+                settings_target = elem.get("settings_target")
+                if settings_target:
+                    compat_info.settings_target.update_values(settings_target)
```

Comment:
``compat_info`` has been created with ``.clone()``, but it means that in some occasions it can be ``None``. Calling ``update_values()`` over it will cause a crash.
I think there is some cases that need to be covered here, for example using the external ``compatibility.py`` plugin over a normal "host" package will crash, so better cover that case too with some extra tests.

Reply:
Done.